web/pages/notifications.tsx

```diff
@@ -69,60 +167,326 @@ export default function Notifications() {
   )
 }
 
-function Notification(props: {
-  currentUser: User
-  notification: Notification
+const setNotificationsAsSeen = (notifications: Notification[]) => {
+  notifications.forEach((notification) => {
+    if (!notification.isSeen)
+      updateDoc(
+        doc(db, `users/${notification.userId}/notifications/`, notification.id),
+        {
+          ...notification,
+          isSeen: true,
+          viewTime: new Date(),
+        }
+      )
+  })
+  return notifications
+}
+
+function GroupNotifications(
+  notifications: Notification[],
+  hideNotificationIds: string[]
+) {
+  // Because hidden notifications won't be rendered, set them to seen here
+  setNotificationsAsSeen(
```

Comment:
Weird to have effects like this within a function to group notifications...
Could you move this out?

Reply:
good thinking!